src/Promitor.Agents.Scraper/Validation/Steps/Sinks/OpenTelemetryCollectorMetricSinkValidationStep.cs

```diff
@@ -33,6 +34,18 @@ public ValidationResult Run()
 
             var errorMessages = new List<string>();
             var collectorUri = openTelemetryCollectorSinkConfiguration.CollectorUri;
+            var collectorProtocol = openTelemetryCollectorSinkConfiguration.Collector.CollectorProtocol;
+
+            // Protocol Validation
+            if (string.IsNullOrWhiteSpace(collectorProtocol))
+            {
+                errorMessages.Add("No Protocol for the OpenTelemetry Collector is configured.");
+            }
+            else if (collectorProtocol != "http" && collectorProtocol != "grpc")
+            {
+                errorMessages.Add($"Configured Protocol ({collectorProtocol}) for the OpenTelemetry Collector should be either 'http' or 'grpc'.");
+            }
```

Comment:
This goes away when using `OtlpExportProtocol`, but let's add some unit tests for the new configuration parsing please

Reply:
I might need some help here but we paired on this and its working I believe?